Source/WebCore/style/MatchedDeclarationsCache.cpp

```diff
@@ -169,7 +177,14 @@ void MatchedDeclarationsCache::add(const RenderStyle& style, const RenderStyle&
     ASSERT(hash);
     // Note that we don't cache the original RenderStyle instance. It may be further modified.
     // The RenderStyle in the cache is really just a holder for the substructures and never used as-is.
-    m_entries.add(hash, Entry { &matchResult, RenderStyle::clonePtr(style), RenderStyle::clonePtr(parentStyle), userAgentAppearanceStyleCopy() });
+    constexpr unsigned maxEntriesPerHash = 4;
+    auto addResult = m_entries.ensure(hash, [&] {
+        Vector<Entry> newBucket;
+        newBucket.reserveCapacity(maxEntriesPerHash);
+        return newBucket;
```

Comment:
It is kinda wasteful in terms of memory to reserve space for 4 entries when if there is often only one (though this is still better than not reserving capacity at all due to Vector growth strategy). Something like `std::array<std::unique_ptr<Entry>, 4>` might be more optimal. But this can be tweaked later.